### PR TITLE
Check struct sizes on the CI instead of during compile-time

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/FTL/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 

## 10

---

Check struct sizes on the CI (in code) instead of during compile-time (using the preprocessor). This should resolve an issue reported on [Discourse](https://discourse.pi-hole.net/t/static-assert-h-error-compiling-from-source-alpine-linux/54915).

It adds an (undocumented) new flag `pihole-FTL --check-structs` that is supposed to be used by the tests. The exit code will tell if everything worked as expected (`bash`-compatible: exit code 0 is success).

### CODE REVIEW

**Comment by dschaper** [comment]
Thanks!

File: src/config.h [hunk 1]
```diff
@@ -17,10 +17,12 @@
...
 // typedef uni32_t
 #include <stdint.h>
-// assert_sizeof
-#include "static_assert.h"
 // struct in_addr, in6_addr
 #include <netinet/in.h>
+// type bool
+#include <stdbool.h>
+// type FILE
+#include <stdio.h>
```
Comment (yubiuser) [inline src/config.h:25]: Why are those two `include` needed now?

File: src/datastructure.h [hunk 2]
```diff
@@ -15,8 +15,9 @@
 
 // enum privacy_level
 #include "enums.h"
-// assert_sizeof
-#include "static_assert.h"
+
+// Definitions like OVERTIME_SLOT
+#include "FTL.h"
```
Comment (yubiuser) [inline src/datastructure.h:20]: Why is this needed now?

File: src/config.h (hunk 1 above)
Comment (DL6ER) [reply src/config.h:25]: We remove the `assert.h` dependency which causes some automatic "behind-the-scenes" includes to not happen. Being more explicit what we want seems a better idea in general.

File: src/datastructure.h (hunk 2 above)
Comment (DL6ER) [reply src/datastructure.h:20]: See https://github.com/pi-hole/FTL/pull/1329#discussion_r857076488

**Comment by DL6ER** [comment]
It's worth pointing out that we will only test struct sizes on platform where we run the tests for (`x86`, `x86-musl`, `i386`). However, this is *not* a drawback as we are checking on both 32bit and a 64bit architectures. (also note that there is *no* difference between `arm`, `mips` and `i386` so everything is still covered).

**Review by yubiuser — APPROVED** [review]
